src/Organizations/OrganizationEdit/OrganizationEdit.js

```diff
@@ -66,32 +88,42 @@ export const OrganizationEdit = ({ match, history, location, mutator }) => {
     [location.search],
   );
 
-  const updateOrganization = useCallback(
-    (values, { getFieldState }) => {
-      const { [BANKING_INFORMATION_FIELD_NAME]: bankingInformation, ...data } = values;
-
-      return mutator.editOrganizationOrg.PUT(data)
-        .then(() => {
-          return manageBankingInformation({
-            initBankingInformation: getFieldState(BANKING_INFORMATION_FIELD_NAME)?.initial,
-            bankingInformation,
-            organization: values,
-          });
-        })
-        .then(() => {
-          setTimeout(cancelForm);
-          showCallout({
-            messageId: 'ui-organizations.save.success',
-            values: { organizationName: data.name },
-          });
-        })
-        .catch(async e => {
-          await handleSaveErrorResponse(intl, showCallout, e);
+  const updateOrganization = useCallback((
+    { [SUBMIT_ACTION_FIELD_NAME]: submitAction, ...values },
+    { getFieldState },
+  ) => {
+    const { [BANKING_INFORMATION_FIELD_NAME]: bankingInformation, ...data } = values;
+
+    return mutator.editOrganizationOrg.PUT(data)
+      .then(() => {
+        return manageBankingInformation({
+          initBankingInformation: getFieldState(BANKING_INFORMATION_FIELD_NAME)?.initial,
+          bankingInformation,
+          organization: values,
         });
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [cancelForm, intl, manageBankingInformation, showCallout],
-  );
+      })
+      .then(() => {
+        showCallout({
+          messageId: 'ui-organizations.save.success',
+          values: { organizationName: data.name },
+        });
+
+        switch (submitAction) {
+          case SUBMIT_ACTION.saveAndKeepEditing:
+            setTimeout(saveAndKeepEditingHandler);
```

Comment:
Do you really need timeouts?

Reply:
I suggest using `restart` for the form to reset its state and display a confirmation modal when attempting to cancel editing with newly "dirty" fields.